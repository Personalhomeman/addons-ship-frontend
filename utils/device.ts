import * as ReactDeviceDetect from 'react-device-detect';
import * as CompareVersions from 'compare-versions';

export const isAndroid = () => ReactDeviceDetect.isAndroid;
export const isIOS = () => ReactDeviceDetect.isIOS;
export const osVersion = () => ReactDeviceDetect.osVersion;
export const mobileModel = () => ReactDeviceDetect.mobileModel;
export const compareVersions = (firstVersion: string, secondVersion: string) => CompareVersions(firstVersion, secondVersion)
