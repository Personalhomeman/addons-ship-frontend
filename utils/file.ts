import fetch from 'node-fetch';

export const uploadFileToS3 = async (file: File | undefined, presignedUrl: string | undefined) => {
  if (!file || !presignedUrl) throw new Error('File or presignedUrl is invalid');

  const options = {
    method: 'put',
    body: file as any
  };

  return fetch(presignedUrl, options).then(resp => {
    if (!resp.ok) throw new Error(`${resp.status}: ${resp.statusText}`);
  });
};
