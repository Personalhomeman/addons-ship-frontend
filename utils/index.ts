import mapKeys from 'lodash/mapKeys';
import mapValues from 'lodash/mapValues';
import camelCase from 'lodash/camelCase';
import snakeCase from 'lodash/snakeCase';
import isPlainObject from 'lodash/isPlainObject';

export const actionTypeCreator = (prefix: string) => (actionType: TemplateStringsArray) => `${prefix}_${actionType}`;

export const camelizeKeys = <T>(input: Object): T =>
  (mapKeys(input, (_, key: string) => camelCase(key)) as unknown) as T;

export const camelizeKeysDeep = <T>(input: Object): T =>
  mapValues(camelizeKeys(input), (value: any) => (isPlainObject(value) ? camelizeKeysDeep(value) : value)) as T;

export const snakifyKeys = <T>(input: Object): T =>
  (mapKeys(input, (_, key: string) => snakeCase(key)) as unknown) as T;

export const snakifyKeysDeep = <T>(input: Object): T =>
  mapValues(snakifyKeys(input), (value: any) => (isPlainObject(value) ? snakifyKeysDeep(value) : value)) as T;
