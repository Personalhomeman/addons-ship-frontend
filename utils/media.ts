import { useState, useEffect } from 'react';

type Handler = (e: MediaQueryListEvent) => void;

const getListeners = (queries: string[], handler: Handler) =>
  queries.map(query => {
    const listener = window.matchMedia(query);

    listener.addListener(handler);

    return listener;
  });

export const mediaQuery = (...rules: string[]) => {
  const mediaRules = rules.map(r => `(min-width: ${r})`);

  const stateHandlers = rules.map(() => useState(false));

  const getStateHandler = (media: string) => {
    const idx = mediaRules.indexOf(media);

    return stateHandlers[idx];
  };

  useEffect(() => {
    const handler = ({ matches, media }: MediaQueryListEvent) => {
      const [, set] = getStateHandler(media);

      set(matches);
    };

    const listeners = getListeners(mediaRules, handler);

    listeners.forEach(({ media, matches }) => {
      const [, set] = getStateHandler(media);
      set(matches);
    });

    return () => listeners.forEach(l => l.removeListener(handler));
  });

  return stateHandlers.map(([isMatch]) => isMatch);
};
