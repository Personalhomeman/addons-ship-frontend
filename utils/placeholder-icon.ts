import { placeholderIcons } from '@/config';
import { ProjectType } from '@/models/app';

export const getIcon = (appIcon?: string, projectType?: ProjectType) => {
  let icon = appIcon;

  if (!icon) {
    icon = placeholderIcons[projectType!] || placeholderIcons.other;
  }

  const suffix = Math.random()
    .toString(16)
    .replace(/^0\./, '');

  return `${icon}?${suffix}`;
};
