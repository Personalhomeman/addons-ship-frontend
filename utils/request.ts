import fetch, { RequestInit } from 'node-fetch';

const getHeaders = (token: string) => ({
  Authorization: token
});

interface RequestParams extends RequestInit {
  url: string;
  token: string;
}

export const request = ({ url, method, token, body }: RequestParams) =>
  fetch(url, {
    method,
    headers: getHeaders(token),
    body
  });

export const get = (url: string, token: string) => request({ method: 'get', url, token });
export const del = (url: string, token: string) => request({ method: 'delete', url, token });
export const patch = (url: string, token: string) => request({ method: 'patch', url, token });
export const post = (url: string, token: string, body: RequestInit['body']) =>
  request({ method: 'post', url, token, body });
export const put = (url: string, token: string, body: RequestInit['body']) =>
  request({ method: 'put', url, token, body });
