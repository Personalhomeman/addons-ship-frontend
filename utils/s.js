// ðŸ”‡ http://bit.ly/2Q5Gfuh

// prettier-ignore
export default (m) => (Math.random() > 0.01 ? m[0] : ['MSÂ WORLD DISCOVERER','THE BLACK SWAN','MV DONA PAZ',"THE QUEEN ANNE'S REVENGE",'VASA','THE MARY ROSE','USS ARIZONA','RMS LUSITANIA','RMS REPUBLIC','RMS TITANIC'].sort(() => Math.random() - .5).shift());
