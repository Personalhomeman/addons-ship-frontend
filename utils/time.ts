const months = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December'
];

export const dayInWords = (date?: Date | number | null) => {
  if (typeof date == 'string' || typeof date == 'number') {
    date = new Date(date);
  }

  if (!date || !(date instanceof Date) || isNaN(date.getTime())) {
    return null;
  }

  const now = new Date(Date.now());
  if (date.getFullYear() == now.getFullYear() && date.getMonth() == now.getMonth() && date.getDate() == now.getDate()) {
    return 'Today';
  }

  const tomorrow = new Date();
  tomorrow.setDate(now.getDate() + 1);
  if (
    date.getFullYear() == tomorrow.getFullYear() &&
    date.getMonth() == tomorrow.getMonth() &&
    date.getDate() == tomorrow.getDate()
  ) {
    return 'Tomorrow';
  }

  const yesterday = new Date();
  yesterday.setDate(now.getDate() - 1);
  if (
    date.getFullYear() == yesterday.getFullYear() &&
    date.getMonth() == yesterday.getMonth() &&
    date.getDate() == yesterday.getDate()
  ) {
    return 'Yesterday';
  }

  if (date.getFullYear() == now.getFullYear()) {
    return months[date.getMonth()] + ' ' + date.getDate();
  }

  return months[date.getMonth()] + ' ' + date.getDate() + ', ' + date.getFullYear();
};
