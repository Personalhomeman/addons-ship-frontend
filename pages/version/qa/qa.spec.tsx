jest.mock('@/utils/media');

import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { mediaQuery } from '@/utils/media';

import QA from '.';

describe('QA tab', () => {
  describe('it renders without errors', () => {
    test('on mobile', () => {
      (mediaQuery as jest.Mock).mockReturnValue([false]);

      const tree = shallowToJson(shallow(<QA />));

      expect(tree).toMatchSnapshot();
    });

    test('on desktop', () => {
      (mediaQuery as jest.Mock).mockReturnValue([true]);

      const tree = shallowToJson(shallow(<QA />));

      expect(tree).toMatchSnapshot();
    });
  });
});
