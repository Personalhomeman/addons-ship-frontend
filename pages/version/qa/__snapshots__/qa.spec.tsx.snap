// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QA tab it renders without errors on desktop 1`] = `
<Flex
  alignChildrenHorizontal="middle"
  direction="vertical"
  paddingVertical="x16"
>
  <Flex
    alignChildrenHorizontal="middle"
    direction="vertical"
    maxWidth="730px"
  >
    <Text
      color="grape-4"
      size="x6"
      weight="bold"
    >
      Feature coming soon! ðŸš€
    </Text>
    <Text
      align="middle"
      color="gray-7"
      margin="x5"
      size="x3"
    >
      We are planning to extend Ship's functionality with more collaboration options for stakeholders in the project (QA, engineers, product managers, designers and so on) and also to support gated releases. Please let us know what you would like to see on this tab.
    </Text>
    <Base
      backgroundColor="grape-1"
      borderRadius="x2"
      padding="x5"
    >
      <Text
        Component="a"
        color="grape-3"
        href="https://discuss.bitrise.io/c/feature-request/ship"
        size="x4"
        target="_blank"
        weight="bold"
      >
        https://discuss.bitrise.io/c/feature-request/ship
      </Text>
    </Base>
  </Flex>
</Flex>
`;

exports[`QA tab it renders without errors on mobile 1`] = `
<Flex
  alignChildrenHorizontal="middle"
  direction="vertical"
  paddingVertical="x16"
>
  <Flex
    alignChildrenHorizontal="middle"
    direction="vertical"
    maxWidth="90%"
  >
    <Text
      color="grape-4"
      size="x4"
      weight="bold"
    >
      Feature coming soon! ðŸš€
    </Text>
    <Text
      align="middle"
      color="gray-7"
      margin="x5"
      size="x3"
    >
      We are planning to extend Ship's functionality with more collaboration options for stakeholders in the project (QA, engineers, product managers, designers and so on) and also to support gated releases. Please let us know what you would like to see on this tab.
    </Text>
    <Base
      backgroundColor="grape-1"
      borderRadius="x2"
      padding="x5"
    >
      <Text
        Component="a"
        color="grape-3"
        href="https://discuss.bitrise.io/c/feature-request/ship"
        size="x3"
        target="_blank"
        weight="bold"
      >
        https://discuss.bitrise.io/c/feature-request/ship
      </Text>
    </Base>
  </Flex>
</Flex>
`;
